File: Java/src/Creature.java
public abstract class Creature extends AquariumObject {
  private final float eatRadius;
  private boolean isMovingRight;

  public Creature(Aquarium _aquarium, float _speed, Vector2 _position, float _eatRadius) {
    super(_aquarium, _speed, _position);
    eatRadius = _eatRadius;
    isMovingRight = true;
  }

  public float getEatRadius() {
    return eatRadius;
  }

  public boolean getIsMovingRight() {
    return isMovingRight;
  }

  public void setIsMovingRight(boolean _isMovingRight) {
    isMovingRight = _isMovingRight;
  }

  public abstract Vector2 eat();
}

File: C++/src/Creature.cpp
#include "Creature.hpp"

Creature::Creature(Aquarium& _aquarium, float _speed, Vector2 _position, float _eatRadius) : AquariumObject(_aquarium, _speed, _position), eatRadius(_eatRadius) {
    setIsMovingRight(true);
}

float Creature::getEatRadius() const {
    return eatRadius;
}

bool Creature::getIsMovingRight() const {
    return isMovingRight;
}

void Creature::setIsMovingRight(bool _isMovingRight) {
    isMovingRight = _isMovingRight;
}

